client/src/features/visualizer-threejs/store/tangle.ts

```diff
@@ -202,5 +209,33 @@ export const useTangleStore = create<TangleState>()(
                 clickedInstanceId,
             }));
         },
+        addToConfirmedBlocksBySlot: (blockId, slot) => {
+            set((state) => {
+                state.confirmedBlocksBySlot.has(slot)
+                    ? state.confirmedBlocksBySlot.get(slot)?.push(blockId)
+                    : state.confirmedBlocksBySlot.set(slot, [blockId]);
+                return {
+                    ...state,
+                    confirmedBlocksBySlot: state.confirmedBlocksBySlot,
+                };
+            });
+        },
+        removeConfirmedBlocksSlot: (slot) => {
+            set((state) => {
+                state.confirmedBlocksBySlot.delete(slot);
+
+                // Cleanup all slots that are lower than the current slot
+                for (const existingSlot of state.confirmedBlocksBySlot.keys()) {
+                    if (existingSlot < slot) {
+                        state.confirmedBlocksBySlot.delete(existingSlot);
+                    }
+                }
```

Comment:
Question for reviewers: 
We can safely remove previous slots when one gets confirmed, right? Like if 592 gets confirmed, we can remove 591, 590...?

Reply:
What does "removing a slot" in this context do ? Do we use that info only to colour ?

Reply:
in this map yes, its like a queue to color the blocks.
I wanted to make a "fallback cleanup" in case something fails we don't store unnecessary stuff

Reply:
Perhaps I need to rename the map 🤔

Reply:
I guess we can safely remove then, if the old one are "used" and removing won't affect the colours.

Reply:
@panteleymonchuk has this happened to you? From what I've read, when a slot is finalized and the blocks are unconfirmed, they become orphans. And since slot 592 should be finished, that means that the block in the slot 590 was removed from the consensus

Reply:
You mean that the operation for getting the confirmation for the finalized block on the slot 590 gets "delayed"? In that case, we should have set the blocks for 590 as finalized already. So, if it's then added to the slot 590 for finalized blocks, if the slot 593 finishes, it will iterate over the map and remove the slots with less index than 593 and remove them, since it should have already finished. I don't think it makes sense to paint a block later when the slot has already finished